mlir/lib/IR/PatternMatch.cpp

```diff
@@ -229,7 +229,10 @@ void RewriterBase::eraseOp(Operation *op) {
       // until the region is empty. (The block graph could be disconnected.)
       while (!r.empty()) {
         SmallVector<Block *> erasedBlocks;
-        for (Block *b : llvm::post_order(&r.front())) {
+        // Some blocks may have invalid successor, use a set including nullptr
+        // to avoid null pointer.
+        std::set<Block *> visited{nullptr};
```

Comment:
std::set is rarely the right choice in LLVM: https://llvm.org/docs/ProgrammersManual.html#picking-the-right-data-structure-for-a-task

Reply:
The post-order iterator template defaults to SmallPtrSet